models/report.py

```diff
@@ -313,7 +314,15 @@ def scan_datastore(self, ds, process_step=None, session=None):
             The session to use for committing the changes to the database.
             If not given, will open a session and close it at the end
             of the function.
+
+            NOTE: it may be better not to provide the external session
+            to this function. That way it will only commit this report,
+            and not also save other objects that were pending on the session.
```

Comment:
I don't think that matters. The internal session only saves the report and exits, it will not make a lot of difference in terms of number of concurrent sessions.